test/RibbonThetaVault.ts

```diff
@@ -1172,42 +1195,75 @@ function behavesLikeRibbonOptionsVault(params: {
 
         await vault.connect(ownerSigner).rollToNextOption();
 
+        let bidMultiplier = 2;
+
         const auctionDetails = await bidForOToken(
           gnosisAuction,
           optionsPremiumPricer,
           assetContract,
           userSigner.address,
-          params.expectedMintAmount,
+          defaultOtokenAddress,
           params.premium,
-          "2"
+          tokenDecimals,
+          bidMultiplier.toString()
         );
 
-        const vaultOTokenBalanceBefore = (
-          await defaultOtoken.balanceOf(vault.address)
-        ).toString();
-        await gnosisAuction
+        assert.equal(
+          (await defaultOtoken.balanceOf(vault.address)).toString(),
+          "0"
+        );
+
+        const assetBalanceBeforeSettle = await assetContract.balanceOf(
+          vault.address
+        );
+
+        let res = await gnosisAuction
           .connect(userSigner)
           .settleAuction(auctionDetails[0]);
-        const vaultOTokenBalanceAfter = (
-          await defaultOtoken.balanceOf(vault.address)
-        ).toString();
 
         assert.isAbove(
-          parseInt(vaultOTokenBalanceAfter),
-          parseInt(vaultOTokenBalanceBefore)
+          parseInt((await defaultOtoken.balanceOf(vault.address)).toString()),
+          parseInt(
+            params.expectedMintAmount
+              .div(bidMultiplier)
+              .mul(params.premiumDiscount.sub(1))
+              .div(1000)
+              .toString()
+          )
+        );
+
+        assert.isAbove(
+          parseInt((await assetContract.balanceOf(vault.address)).toString()),
+          parseInt(
+            (
+              (assetBalanceBeforeSettle.add(BigNumber.from(auctionDetails[2])) *
+                99) /
+              100
+            ).toString()
+          )
         );
 
-        const collateralBalanceBefore = (
-          await assetContract.balanceOf(vault.address)
-        ).toString();
+        const lockedAmountBeforeBurn = (await vault.vaultState()).lockedAmount;
+        const assetBalanceAfterSettle = await assetContract.balanceOf(
+          vault.address
+        );
         vault.connect(ownerSigner).burnRemainingOTokens();
-        const collateralBalanceAfter = (
-          await assetContract.balanceOf(vault.address)
-        ).toString();
+        const assetBalanceAfterBurn = await assetContract.balanceOf(
+          vault.address
+        );
 
         assert.isAbove(
-          parseInt(collateralBalanceAfter),
-          parseInt(collateralBalanceBefore)
+          parseInt(assetBalanceAfterBurn.toString()),
+          parseInt(
+            assetBalanceAfterSettle
+              .add(
+                lockedAmountBeforeBurn
```

Comment:
similar to how [here](https://github.com/ribbon-finance/ribbon-v2/pull/25/files#r658355771) the person gets more tokens, this means we end up with _less_ tokens to burn, so a smaller balance afterwards (~3 less)